vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java

```diff
@@ -402,10 +405,27 @@ public ByteBuf readBlocking() throws IOException {
         }
     }
 
+    private void failOnError() throws IOException {
+        if (timedOut) {
+            close();
+            throw new IOException("The client failed to write request in time");
+        }
+        if (readError != null) {
+            throw new IOException(readError);
+        }
+    }
+
     @Override
     public void close() {
         synchronized (request.connection()) {
-            request.connection().close();
+            switch (request.version()) {
```

Comment:
I've been told that's the proper way of closing the connection but maybe there's some special case here where we don't want to do it this way?

Reply:
currently it seems to me the most appropriate, in the future in vertx we might implement reset() as a connection close for HTTP/1, for now it will simply do nothing.

Reply:
connection close on HTTP/2 would close  all streams